pkg/services/dashboards/models.go

```diff
@@ -262,6 +262,7 @@ type GetDashboardQuery struct {
 	OrgID     int64
 
 	IncludeDeleted bool // only supported when using unified storage
+	IncludeDTO     bool // Add DTO subresource. Used for K8s dashboards when large object support is enabled.
```

Comment:
```suggestion
```
does this need to be an option?  It feels confusing to explain when/where you could skip the dto vs just always calling it

Reply:
Thanks for the review! Without it, we have a [circular dependency](https://github.com/grafana/grafana/blob/b9e5ccb25d62feb4cc9e295ff123f6d5759c4a74/pkg/services/dashboards/accesscontrol.go#L157). GetDashboard --> DTOConnector.Connect --> NewDashboardUIDScopeResolver --> GetDashboard

More context in [this thread](https://raintank-corp.slack.com/archives/C06JFJ7H3GC/p1740133457706959?thread_ts=1740129950.609289&cid=C06JFJ7H3GC).

Another option would be only evaluating the [Access field](https://github.com/grafana/grafana/blob/b9e5ccb25d62feb4cc9e295ff123f6d5759c4a74/pkg/registry/apis/dashboard/sub_dto.go#L173) in Connect when necessary. Would need to figure out how to pass a field to indicate that to Connect though. Do you think that's preferable?

Reply:
yes i think this is cleaner.

In https://github.com/grafana/grafana/blob/main/pkg/registry/apis/dashboard/sub_dto.go#L126 we can now skip access lookup

The challenge is that you would need to use the raw HTTP client rather than the generic k8s flavor because that does not accept arbitrary HTTP parameters

Reply:
Thanks again for the review. I took a look at implementing your suggestion. It makes sense but will take more time than I had anticipated. I would like to prioritise getting large object support deployed in dev first. Can I convince you to sign off on the current implementation and I’ll add the raw HTTP client task as a later refactor? It should not introduce any breaking/user-facing changes from what I can tell and can be a behind-the-scenes improvement.